### DISCUSSION TITLE
some array problems.

### DISCUSSION THREAD

**Discussion by starktwo** [discordjs/discord.js#9189] [Q&A] [answered]
### Which package is this bug report for?

discord.js

### Issue description

Hi, I do not know if I have made a discord.js error or something caused by me recently. Although he was string, he did not give the role without Array.

![image](https://user-images.githubusercontent.com/51216503/222316736-8ac45e3a-9060-4c72-9669-b3f90675f531.png)
![image](https://user-images.githubusercontent.com/51216503/222316883-b52ad772-e4a9-4419-86be-07d4335451de.png)

### Code sample

```typescript
const unRegRole = "901530734991384646",
module.exports = {
  conf: {
    cmd: "rolsuz",
    aliases: ["rolsüz"],
    enabled: true,
    cmdPerms: ["cmdOwner"]

  },
  run: async (client, message, args) => {
    let rolsuz = message.guild.members.cache.filter(a => a.roles.cache.size <= 1)

    if (rolsuz.size == 0) return message.channel.send(`\`\`\`Sunucumuzda rolsüz üye bulunmamakta.\`\`\``)

    await rolsuz.forEach(member => {
      member.roles.add([unRegRole]).catch(() => { })
    });

    await message.channel.send(`\`\`\`Herhangi bir rolü olmayan ${rolsuz.size} kişiye kayıtsız rolü verildi.\`\`\``)

  }

};
```

### Package version

14.5.0

### Node.js version

v18.14.0

### Operating system

Windows 11

### Priority this issue should have

High (immediate attention needed)

### Which partials do you have configured?

User, Channel, GuildMember, Message, Reaction, GuildScheduledEvent, ThreadMember

### Which gateway intents are you subscribing to?

Guilds, GuildMembers, GuildModeration, GuildEmojisAndStickers, GuildIntegrations, GuildWebhooks, GuildInvites, GuildVoiceStates, GuildPresences, GuildMessages, GuildMessageReactions, GuildMessageTyping, DirectMessages, DirectMessageReactions, DirectMessageTyping, MessageContent, GuildScheduledEvents, AutoModerationConfiguration, AutoModerationExecution

**Comment by Syjalo** [comment]
Not all members are cached, so that it doesn't add roles sometimes. It's better to implement that feature in the `guildMemberAdd` event.

**Reply by starktwo** [reply]
Not only is your friend on this command, but even if he attracts a normal user correctly, it is the same when you create a command to give a role normally.

**Comment by NothingHollow** [comment, marked as answer]
If the cache doesn't provide most members in the cache, try fetching them.

**Reply by starktwo** [reply]
I already use what you say, and I use it all as a cache. Ready at the event

**Reply by jaw0r3k** [reply]
As they said, you have to fetch the members ([GuildMemberManager#fetch()](https://discord.js.org/#/docs/discord.js/main/class/GuildMemberManager?scrollTo=fetch))

**Reply by starktwo** [reply]
> Dedikleri gibi, üyeleri getirmelisiniz ( [GuildMemberManager#fetch()](https://discord.js.org/#/docs/discord.js/main/class/GuildMemberManager?scrollTo=fetch) )

Even when I do not cache the member, the member information falls on the console when the member prints the data to the console.

**Reply by almostSouji** [reply]
That's not really possible. `<Guild>.members.fetch()` will return a Promise, once that's resolved the client has received all members of the guild. If you then iterate over them and add roles, it will add them eventually. 
Depending on the size of the guild, that is a very large operation and might take hours to complete due to rate limits.
